localstack-core/localstack/services/lambda_/event_source_mapping/pipe_loggers/pipe_logger.py

```diff
@@ -55,14 +55,31 @@ def filter_message(self, message: dict) -> dict:
             "awsRequest",
             "awsResponse",
         ]
-        fields_to_remove = (
-            []
-            if self.include_execution_data == [IncludeExecutionDataOption.ALL]
-            else execution_data_fields
-        )
-        filtered_message = {
-            key: value for key, value in message.items() if key not in fields_to_remove
-        }
+        fields_to_include = [
+            "resourceArn",
+            "timestamp",
+            "executionId",
+            "messageType",
+            "logLevel",
+        ]
+        error_fields_to_include = [
+            "message",
+            "httpStatusCode",
+            "awsService",
+            "requestId",
+            "exceptionType",
+            "resourceArn",
+        ]
+
+        if self.include_execution_data == [IncludeExecutionDataOption.ALL]:
+            fields_to_include.extend(execution_data_fields)
+
+        filtered_message = {key: message[key] for key in fields_to_include if key in message}
```

Comment:
```suggestion
        filtered_message = {key: value for key, value in message.items() if key in fields_to_include}
```
Could we do it this way? I think this is more straightforward to read, as it is pure filtering. By also converting the fields_to_include lists to sets, we could even reduce time complexity by a bit (although it honestly does not matter in this case).

Reply:
Making the above sets makes sense. Just actually did a quick `timeit` (thinking it wouldn't make much of a difference) and it turns out it very much does even with so few elements 🫠:
```python
>>> import timeit
>>> timeit.timeit(stmt='-1 in a', setup='a = list(range(10))', number=1000)
0.0001385409850627184
>>> timeit.timeit(stmt='-1 in a', setup='a = set(range(10))', number=1000)
4.804192576557398e-05
```

I opted for the above approach since we are going to be reworking this filtering API in the coming weeks in anycase. Happy to make this more readable though.

Reply:
You are right, it definitely makes sense on that order of magnitude, I was talking more about the LS context, where it is nothing compared to some other overhead we see :) Still, we should always strive to increase performance)